src/dataflow/src/sink/kafka.rs

```diff
@@ -574,3 +581,161 @@ where
 
     Box::new(KafkaSinkToken { shutdown_flag })
 }
+
+/// An operator for encoding Rows as Avro that can yield, and doesn't have to process all its in
+/// put in one invocation.
+fn avro_encode_stream<G>(
+    input_stream: &Stream<G, ((Option<Row>, Option<Row>), u64, isize)>,
+    encoder: Encoder,
+    key_schema_id: Option<i32>,
+    value_schema_id: i32,
+    fuel: usize,
+    sink_name: String,
+) -> Stream<G, ((Option<Vec<u8>>, Option<Vec<u8>>), u64, isize)>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let name = format!("{}-avro_encode", sink_name);
+
+    let mut builder = OperatorBuilder::new(name, input_stream.scope());
+    let mut input = builder.new_input(&input_stream, Pipeline);
+    let (mut output, output_stream) = builder.new_output();
+    builder.set_notify(false);
+
+    let activator = input_stream
+        .scope()
+        .activator_for(&builder.operator_info().address[..]);
+
+    let mut stash: HashMap<Capability<Timestamp>, Vec<_>> = HashMap::new();
+    let mut vector = Vec::new();
+    let mut encode_logic = move |input: &mut FrontieredInputHandle<
+        Timestamp,
+        ((Option<Row>, Option<Row>), Timestamp, Diff),
```

Comment:
I don't know if you can do anything about this but the formatting here is hard to read :/

Reply:
Very strange, yes. But this is what rustfmt will give me. 😞